src/app/services/post.service.ts

```diff
@@ -0,0 +1,31 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { Post} from 'src/types/posts';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class PostService {
+
+  private _posts: Post[];
+
+  posts$: BehaviorSubject<Post[]> = new BehaviorSubject(null);
+
+  constructor() { }
+
+  setPosts(posts: Post[]): void {
+    this._posts = posts;
+    this.posts$.next(posts);
+  }
+
+  getAllPosts(): Observable<Post[]> {
+    // return this._posts;
```

Comment:
You can directly subscribe on the public property `posts$`, so this potentially doesn't have to be here at all